pentaho-spring-dm-extender/pom.xml

```diff
@@ -17,22 +17,34 @@
     <exam.version>4.1.0</exam.version>
   </properties>
   <dependencies>
-    <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-beans</artifactId>
-    </dependency>
     <dependency>
       <groupId>org.springframework</groupId>
       <artifactId>spring-core</artifactId>
     </dependency>
+
+    <!-- region - Needed for compilation -->
     <dependency>
       <groupId>org.springframework</groupId>
       <artifactId>spring-context</artifactId>
+      <scope>compile</scope>
+    </dependency>
+    <!-- endregion -->
+
+    <dependency>
+      <groupId>org.springframework.osgi</groupId>
+      <artifactId>spring-osgi-core</artifactId>
+      <version>${spring-osgi.version}</version>
+      <exclusions>
+        <exclusion>
+          <artifactId>*</artifactId>
```

Comment:
Why the exclusions? Also, what in particular from OSGi is being used? Why not the standard spring-core artifact?

Reply:
Why the exclusions? -> There were some spring dependencies, like beans that were being including in wrong versions due to this dependency
Also, what in particular from OSGi is being used? -> Multiple imports on ApplicationContextCreator.java and PentahoOsgiBundleXmlApplicationContext.java. To understand in detail what is being used and why not being used spring-core, maybe a deeper analysis will be required. I was trying to limit the impact of changes that I was making.